engine/runtime-parser/src/main/java/org/enso/compiler/core/ir/type/Set.java

```diff
@@ -0,0 +1,221 @@
+    /**
+     * @param left the left operand
+     * @param right the right operand
+     * @param identifiedLocation the source location that the node corresponds to
+     * @param passData the pass metadata associated with this node
+     */
+    @GenerateFields
+    public Subsumption(
+        @IRChild Expression left,
+        @IRChild Expression right,
+        IdentifiedLocation identifiedLocation,
+        MetadataStorage passData) {
+      super(left, right, identifiedLocation, passData);
+    }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    public static String name() {
+      return "<:";
+    }
+
+    public Builder copyBuilder() {
+      return new Builder(this);
+    }
+
+    @Override
+    public String showCode(int indent) {
+      return "(" + left().showCode(indent) + " <: " + right().showCode(indent) + ")";
+    }
+  }
+
+  /** The typeset equality judgement {@code ~}. */
+  @GenerateIR(interfaces = {Set.class, IRKind.Primitive.class})
+  final class Equality extends SetEqualityGen {
+    @GenerateFields
+    public Equality(
+        @IRChild Expression left,
+        @IRChild Expression right,
+        IdentifiedLocation identifiedLocation,
+        MetadataStorage passData) {
+      super(left, right, identifiedLocation, passData);
+    }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    public static String name() {
+      return "~";
+    }
+
+    public Builder copyBuilder() {
+      return new Builder(this);
+    }
+
+    @Override
+    public String showCode(int indent) {
+      return "(" + left().showCode(indent) + " ~ " + right().showCode(indent) + ")";
+    }
+  }
+
+  /** The typeset concatenation operator {@code ,}. */
+  @GenerateIR(interfaces = {Set.class, IRKind.Primitive.class})
+  final class Concat extends SetConcatGen {
+    @GenerateFields
+    public Concat(
+        @IRChild Expression left,
+        @IRChild Expression right,
+        IdentifiedLocation identifiedLocation,
+        MetadataStorage passData) {
+      super(left, right, identifiedLocation, passData);
+    }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    public static String name() {
+      return ";";
+    }
+
+    public Builder copyBuilder() {
+      return new Builder(this);
+    }
+
+    @Override
+    public String showCode(int indent) {
+      return "(" + left().showCode(indent) + "; " + right().showCode(indent) + ")";
+    }
+  }
+
+  /** The typeset union operator {@code |}. */
+  @GenerateIR(interfaces = {Set.class, IRKind.Primitive.class})
+  final class Union extends SetUnionGen {
+    @GenerateFields
+    public Union(
+        @IRChild List<Expression> operands,
+        IdentifiedLocation identifiedLocation,
+        MetadataStorage passData) {
+      super(operands, identifiedLocation, passData);
+    }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    public static String name() {
```

Comment:
- having static method returning constant is not very Java-like pattern
- static constant field would be more natural
```suggestion
    public static final String NAME = "|";
```
- and maybe it could help with [switch over strings](https://github.com/enso-org/enso/pull/14299#discussion_r2626931640)
- as the compiler will know that `NAME` is a constant
- while it cannot know that `name()` is a constant and refuses to use such values as `case` patterns

Reply:
My bad. Reintroduced in d5682fe47a4068ca96b530c107f74334570fd9ee - it really works.